### PR TITLE
Payara 1122 store a number of historical traces in memory for future reporting

### PR DESCRIPTION
**Pull request by mulderbaba**

### CODE REVIEW

**Comment by mulderbaba** [comment]
jenkins test please

File: nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/HistoricRequestEventStore.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    private SortedSet<HistoricRequestEvent> historicStore;
+
+    void initialize(int storeSize) {
+        historicStore = Collections.synchronizedSortedSet(new BoundedTreeSet<HistoricRequestEvent>(storeSize));
+
+        if (hzCore.isEnabled()) {
+            instance = hzCore.getInstance();
+            String instanceName = serverEnv.getInstanceName();
+            IMap<String, SortedSet<HistoricRequestEvent>> map
```
Comment (mulderbaba) [inline nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/HistoricRequestEventStore.java]: @smillidge is this the proper approach to store traces into the clustered store under the instance name?

File: nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/configuration/RequestTracingServiceConfiguration.java [hunk 2]
```diff
@@ -42,6 +68,15 @@ and Distribution License("CDDL") (collectively, the "License").  You
     String getThresholdUnit();
     void setThresholdUnit(String value) throws PropertyVetoException;
 
+    @Attribute(defaultValue = "false", dataType = Boolean.class)
+    String getHistoricalTraceEnabled();
+    void setHistoricalTraceEnabled(String value) throws PropertyVetoException;
+
+    @Attribute(defaultValue = "20", dataType = Integer.class)
```
Comment (mulderbaba) [inline nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/configuration/RequestTracingServiceConfiguration.java]: @smillidge should we go bigger on the default size?

File: nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/HistoricRequestEventStore.java (hunk 1 above)
Comment (smillidge) [inline nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/HistoricRequestEventStore.java]: Yes look about right except ```java instance.getMap(HISTORIC_REQUEST_EVENT_STORE);``` could return null

File: nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/configuration/RequestTracingServiceConfiguration.java (hunk 2 above)
Comment (smillidge) [inline nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/configuration/RequestTracingServiceConfiguration.java]: I'm fine with 20.

**Pull request closed without merging**